echo_server.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import unicode_literals
+import socket
+
+
+def response_ok(uri):
+    response = []
+    response.append("HTTP/1.1 200 OK")
+    response.append("Content-Type = text/html; charset=utf-8")
+    response.append("")
+    response.append(uri)
+
+    response = "\r\n".join(response).encode("utf-8")
+    return response
+
+
+def response_error(error, error_msg):
+    response = []
+    response.append("HTTP/1.1 {} {}".format(error, error_msg))
+    response.append("")
+    response.append("{} {}".format(error, error_msg))
+    response = "\r\n".join(response).encode("utf-8")
+
+    return response
+
+
+def parse_request(request):
+    first_line = request.splitlines()[0]
+    first_line = first_line.split(" ")
+
+    if first_line[0] == "GET":
+        if first_line[2] == "HTTP/1.1":
+            return response_ok(first_line[1])
+        else:
+            return response_error(505, "Protocol must be HTTP/1.1")
+    else:
+        return response_error(405, "Method must be GET")
```

Comment:
This function is intended to return the URI so that it can be processed later.  Here you are generating responses within this function.  Don't do that.  Raise a Python error or return the URI.